### ISSUE TITLE
FTBFS with gcc 9.x: 'in_range' is not a member of 'std'

### ISSUE THREAD

**Issue by shastah** [rpm-software-management/dnf5#1167]
Trying to compile dnf5 under gcc 9.x (tried 9.2.0, 9.5.0) I'm getting this
```
In file included from /home/rpm/rpmbuild/BUILD/common/utils/string.cpp:21:
/home/rpm/rpmbuild/BUILD/common/utils/string.hpp: In function 'std::string libdnf5::utils::string::format_epoch(T)':
/home/rpm/rpmbuild/BUILD/common/utils/string.hpp:115:14: error: 'in_range' is not a member of 'std'
  115 |     if (std::in_range<time_t>(epoch_num)) {
      |              ^~~~~~~~
/home/rpm/rpmbuild/BUILD/common/utils/string.hpp:115:29: error: expected primary-expression before '>' token
  115 |     if (std::in_range<time_t>(epoch_num)) {
      |                             ^
```
Compiler info
```
# gcc --version
gcc (GCC) 9.2.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

# g++ --version
g++ (GCC) 9.2.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by ppisar** [comment]
According to libstdc++-v3/ChangeLog-2020, std::in_range() was added on 2020-07-07. According to <https://gcc.gnu.org/releases.html>, GCC 9.2 was released on 2019-08-12.

According to <https://en.cppreference.com/w/cpp/utility/in_range> std::range is implemented if 
`__cpp_lib_integer_comparison_functions` feature macro is at least `202002L`. According to <https://en.cppreference.com/w/cpp/feature_test#cpp_lib_integer_comparison_functions> the macro was introduced in [P0586R2](https://wg21.link/P0586R2) draft. According to <https://gcc.gnu.org/onlinedocs/libstdc++/manual/status.html#status.iso.2020> that "Safe integral comparisons" draft was fully implemented in GCC 10.1.

I worry you need to use a newer compiler.

Thanks for pointing to the minimal required version. We will probably resolve this issue be constraining a minimal version of GCC on dnf5.spec and enhancing CMake build script to check for __cpp_lib_integer_comparison_functions >= 202002L macro value.

**Comment by ppisar** [comment]
I checked GCC sources: gcc-10.0.1-20200118 provides neither std::in_range(), nor __cpp_lib_integer_comparison_functions macro. Contrary, gcc-10.0.1-20200501 provides both of them.
